packages/legacy/core/App/screens/PINCreate.tsx

```diff
@@ -210,19 +244,46 @@ const PINCreate: React.FC<PINCreateProps> = ({ setAuthenticated }) => {
             <AlertModal
               title={modalState.title}
               message={modalState.message}
-              submit={() => setModalState({ ...modalState, visible: false })}
+              submit={() => {
+                if (modalState.onModalDismiss) {
+                  modalState.onModalDismiss()
+                }
+                setModalState({ ...modalState, visible: false, onModalDismiss: undefined })
+              }}
             />
           )}
         </View>
         <View style={style.controlsContainer}>
           <Button
-            title={t('PINCreate.CreatePIN')}
-            testID={testIdWithKey('CreatePIN')}
-            accessibilityLabel={t('PINCreate.CreatePIN')}
+            title={updatePin ? t('PINCreate.ChangePIN') : t('PINCreate.CreatePIN')}
+            testID={updatePin ? testIdWithKey('PINCreate.ChangePIN') : testIdWithKey('CreatePIN')}
+            accessibilityLabel={updatePin ? t('PINCreate.ChangePIN') : t('PINCreate.CreatePIN')}
             buttonType={ButtonType.Primary}
             disabled={!continueEnabled}
             onPress={async () => {
```

Comment:
Can we possibly move this function out of the JSX and into a variable in the component logic? Would be easier to read. Not a must, just a nice to have